qadence/pasqal_cloud_connection.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import json
+import time
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Optional
+
+from pasqal_cloud import SDK
+from pasqal_cloud import Workload as WorkloadResult
+from torch import Tensor
+
+from qadence import AbstractBlock, BackendName, QuantumCircuit, QuantumModel, serialize
+
+
+class ResultType(Enum):
+    RUN = "run"
+    SAMPLE = "sample"
+    EXPECTATION = "expectation"
+
+
+@dataclass(frozen=True)
+class WorkloadSpec:
+    """Specification of a workload to be executed on Pasqal Cloud.
+
+    This data class defines a single workload specification that is to be executed on Pasqal's
+    cloud platform.
+
+    Args:
+        circuit: The quantum circuit to be executed.
+        backend: The backend to execute the workload on. Not all backends are available on the
+            cloud platform. Currently the supported backends are `BackendName.PYQTORCH` and
+            `"emu_c"`.
```

Comment:
```suggestion
            `BackendName.EMUC`.
```
Although I am wondering if we should mention a private emulator in a public repo.